### PR TITLE
Refactor/OmegaId

### PR DESCRIPTION
**Pull request by theodorklauritzen**
This reduces the size of the OmegaID QR code a lot. However the solution is maybe not that elegant.

### CODE REVIEW

File: src/services/omegaid/compression.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        const payload = decompressPayload(rawDataSplit[0])
+
+        const signature = bigIntToBase64(rawDataSplit[1])
+
+        return {
+            success: true,
+            data: `${headerB64String}.${payload}.${signature}`,
+        }
+    } catch (e) {
+        console.error(e)
```
Comment (JohanHjelsethStorstad) [inline src/services/omegaid/compression.ts:105]: console.log

File: src/services/omegaid/compression.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import { JWT_ISSUER } from '@/auth/ConfigVars'
+import type { ActionReturn, ActionReturnError } from '@/actions/Types'
+import type { OmegaIdJWT } from '@/services/omegaid/Types'
+
+
```
Comment (JohanHjelsethStorstad) [inline src/services/omegaid/compression.ts:5]: Maybe document the JWT -> QR compression on the top

File: src/jwt/parseJWTClient.ts [hunk 3]
```diff
@@ -74,9 +74,6 @@ export async function parseJWT(token: string, publicKey: string, timeOffset: num
         const payload = readJWTPayload(token)
 
         if (!(
-            typeof payload.usrnm === 'string' &&
```
Comment (JohanHjelsethStorstad) [inline src/jwt/parseJWTClient.ts:77]: I think this file is wrongly placed. First I think the function should be called parseJWTClient and this parse is specific to omegaId so why is it placed with the general jwt stuff. I think this should be split: one part is. in jwt libary it verifies and parses jwt on client. One part decodes and then uses this function - this should be placed with omegaId functionality.

**Review by JohanHjelsethStorstad — CHANGES_REQUESTED** [review]

**Comment by theodorklauritzen** [comment]
Hmmm, the main branch had a bug with a circular dependency in Vevenadapter, this broke feide login in the development environment. The fix for now is to not use the method handler, this is against the convention, and we should maybe discuss a more permanent solution.

**Pull request closed without merging**